core/iwasm/libraries/lib-wasi-threads/lib_wasi_threads_wrapper.c

```diff
@@ -31,16 +39,43 @@ typedef struct {
 static int32
 allocate_thread_id()
 {
-    static int32 thread_id = 0;
-
     int32 id;
 
     os_mutex_lock(&thread_id_lock);
-    id = thread_id++;
+    if (avail_tids.pos < 0) {
+        // Resize stack and push new thread ids
+        uint32 old_size = avail_tids.size;
+        avail_tids.size *= 2;
+        avail_tids.pos = old_size - 1;
+        avail_tids.ids = (int32 *)wasm_runtime_realloc(
+            avail_tids.ids, avail_tids.size * sizeof(int32));
```

Comment:
check overflow?

Reply:
similar here
```
size_t realloc_size = avail_tids.size * sizeof(int32);
if (realloc_size / sizeof(int32) != avail_tids.size) { ... }
```

Reply:
if you want to use them, i guess you need to introduce macros with plain C fallback.